api/v1alpha1/aws_node_types.go

```diff
@@ -49,21 +57,34 @@ func (AdditionalSecurityGroup) VariableSchema() clusterv1.VariableSchema {
 	}
 }
 
-func (AWSNodeSpec) VariableSchema() clusterv1.VariableSchema {
+func (a AWSNodeSpec) VariableSchema() clusterv1.VariableSchema {
 	return clusterv1.VariableSchema{
 		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
 			Description: "AWS Node configuration",
 			Type:        "object",
 			Properties: map[string]clusterv1.JSONSchemaProps{
 				"iamInstanceProfile":       IAMInstanceProfile("").VariableSchema().OpenAPIV3Schema,
-				"instanceType":             InstanceType("").VariableSchema().OpenAPIV3Schema,
+				"instanceType":             a.InstanceType.VariableSchema().OpenAPIV3Schema,
 				"ami":                      AMISpec{}.VariableSchema().OpenAPIV3Schema,
 				"additionalSecurityGroups": AdditionalSecurityGroup{}.VariableSchema().OpenAPIV3Schema,
 			},
+			Required: []string{"instanceType"},
 		},
 	}
 }
 
+func NewAWSControlPlaneNodeSpec() *AWSNodeSpec {
+	return &AWSNodeSpec{
+		InstanceType: ptr.To(AWSControlPlaneInstanceType),
+	}
+}
+
+func NewAWSWorkerNodeSpec() *AWSNodeSpec {
+	return &AWSNodeSpec{
+		InstanceType: ptr.To(AWSWorkerInstanceType),
+	}
+}
+
```

Comment:
nit: maybe move this right under the `AWSNodeSpec` struct to make it easier to spot this constructor?

Reply:
done